generator/generator.go

```diff
@@ -64,29 +65,36 @@ func (g *generator) BatchOperations(logger lager.Logger) (map[string]operationq.
 	evacuatingLRPs := make(map[string]models.ActualLRP)
 	tasks := make(map[string]*models.Task)
 
-	errChan := make(chan error, 3)
-
 	logger.Info("getting-containers-lrps-and-tasks")
+
+	var wg sync.WaitGroup
+	wg.Add(3)
+	var groupErr error
+
 	go func() {
+		defer wg.Done()
+
 		foundContainers, err := g.executorClient.ListContainers(logger)
 		if err != nil {
 			logger.Error("failed-to-list-containers", err)
-			err = fmt.Errorf("failed to list containers: %s", err.Error())
+			groupErr = fmt.Errorf("failed to list containers: %s", err.Error())
+			return
```

Comment:
yes, I think skip the following for loop is good.
even if an error is returned and foundContainer isn't empty,
because the code logic is once an error in the three go routines,  the error would be returnd at 
[https://github.com/andyliuliming/rep/blob/ce1e26df877316b804a23e96f419ff59ba3f7a60/generator/generator.go#L133](url)

and the local variables "containers" "instanceLRPs"  "evacuatingLRPs"  "tasks" would not be used. so the three loops is useless once the error happens.

Reply:
reverted this, and only use multi error now.